t3/gpu_baseline_faiss.py

```diff
@@ -0,0 +1,592 @@
+
+
+    #GW ds = DATASETS[args.dataset]()
+    ds = DATASETS[args.dataset]
+
+    if args.search:
+        # Do a sanity check before loading a large index file
+        q = ds.get_queries()
+        #range search- r, i, g = ds.get_groundtruth()
+        i, g = ds.get_groundtruth()
+
+    if args.build:
+        # Do a sanity check before training large dataset
+        d = ds.get_dataset()
+        for xblock in ds.get_dataset_iterator(bs=10):
+            break
+
+    if args.search:
+
+        # Connect to ipmicap server and do a test
+        power_capture("192.168.99.123:1234:-1")
+        power_capture.ping()
+        capid = power_capture.start()
+        time.sleep(2)
+        power_cons = power_capture.stop(capid, all_stats=True)
+        if power_cons==-1: raise Exception("Power capture failed.")
+        print("Power sensor test passed=", power_cons)
+
+        # Initialize the data files
+        init_data(args)
+
+    nq, d = ds.nq, ds.d
+    nb, d = ds.nq, ds.d
+
+    if args.prepare:
+        print("downloading dataset...")
+        ds.prepare()
+        print("dataset ready")
+
+    if not (args.build or args.search):
+        return
+
+    if args.build:
+        print("build index, key=", args.indexkey)
+        index = build_index(args, ds)
+    else:
+        print("reading", args.indexfile)
+        index = faiss.read_index(args.indexfile)
+
+    index_ivf, vec_transform = unwind_index_ivf(index)
+    if vec_transform is None:
+        vec_transform = lambda x: x
+
+    if index_ivf is not None:
+        print("imbalance_factor=", index_ivf.invlists.imbalance_factor())
+
+    if args.no_precomputed_tables:
+        if isinstance(index_ivf, faiss.IndexIVFPQ):
+            print("disabling precomputed table")
+            index_ivf.use_precomputed_table = -1
+            index_ivf.precomputed_table.clear()
+
+    if args.indexfile:
+        print("index size on disk: ", os.stat(args.indexfile).st_size)
+
+    print("current RSS:", faiss.get_mem_usage_kb() * 1024)
+
+    precomputed_table_size = 0
+    if hasattr(index_ivf, 'precomputed_table'):
+        precomputed_table_size = index_ivf.precomputed_table.size() * 4
+
+    print("precomputed tables size:", precomputed_table_size)
+
+    if args.search:
+
+        if args.searchthreads == -1:
+            print("Search threads:", faiss.omp_get_max_threads())
+        else:
+            print("Setting nb of threads to", args.searchthreads)
+            faiss.omp_set_num_threads(args.searchthreads)
+
+        if args.parallel_mode != -1:
+            print("setting IVF parallel mode to", args.parallel_mode)
+            index_ivf.parallel_mode
+            index_ivf.parallel_mode = args.parallel_mode
+
+        if args.searchparams == ["autotune"]:
+            run_experiments_autotune(ds, index, args)
+        else:
+            run_experiments_searchparams(ds, index, args ) #, 96237 )
+
+if __name__ == "__main__":
+    main()
```

Comment:
Full support for leaving this baseline in, but we should add a comment that this is stand-alone and doesn't use the framework (very much).

Reply:
Actually, i'm fine removing it.  It would just cause unnecessary confusion.